### DISCUSSION TITLE
Add a button that scrolls the page back to top

### DISCUSSION THREAD

**Discussion by dev-palwar** [codu-code/codu#581] [Ideas]
## Expected Behavior
The button should have the functionality to smoothly scroll the page back to the top upon being clicked.

## Current Behavior
Right now, we have to manually scroll up the page, which can be a hassle, especially when reading blogs.

**Comment by NiallJoeMaher** [comment]
Any ideas on where something like this should be added or examples of blogs with it?

**Comment by dev-palwar** [comment]
Yeah, something like this
![Screenshot 2023-10-14 at 14-52-22 Create a workflow to display your latest articles on your GitHub profile](https://github.com/codu-code/codu/assets/102988552/3596c259-ad1a-408f-81a8-0f212d7d442b)

A button on the right side of the page similar to the one homepage that when clicked smoothly scrolls the page to the top

**Comment by MadhavKrishanGoswami** [comment]
I think I can work on this, just one thing where would I get the icon from?

**Comment by NiallJoeMaher** [comment]
@MadhavKrishanGoswami I'm trying to mock up something for the mobile design and then you can take it. 🦾

**Comment by akashsiddamshetty** [comment]
@dev-palwar @NiallJoeMaher  check this below screen shot i have created this button to scroll to top smoothly. I want to make pr. can you guys add hacktoberfest labels to this issue #542


<img width="1582" alt="Screenshot 2023-10-18 at 1 59 09 PM" src="https://github.com/codu-code/codu/assets/80561904/02a9bba2-8caf-445f-b5e3-2d5a606ada2a">

<img width="1582" alt="Screenshot 2023-10-18 at 2 07 37 PM" src="https://github.com/codu-code/codu/assets/80561904/3cd7a098-daaa-4081-a20d-affcc0877423">

**Comment by Bitnagar** [comment]
@akashsiddamshetty Great job! The icon looks good. IMO we should not use gray color because it looks dull and not visually appealing. We can try sticking to the design in 2nd screenshot or else we can use gradient color that is in our theme.

**Comment by akashsiddamshetty** [comment]
> @akashsiddamshetty Great job! The icon looks good. IMO we should not use gray color because it looks dull and not visually appealing. We can try sticking to the design in 2nd screenshot or else we can use gradient color that is in our theme.

@Bitnagar @NiallJoeMaher @dev-palwar  check this out. 
<img width="1582" alt="Screenshot 2023-10-18 at 3 51 45 PM" src="https://github.com/codu-code/codu/assets/80561904/db47de9e-6461-4e87-ba5d-f2c148b03c2d">

**Comment by NiallJoeMaher** [comment]
The problem is when we are on mobile. I just haven't seen a pattern on mobile I like. 😅 Any examples of sites doing it well?

**Comment by NiallJoeMaher** [comment]
Moved to discussion, I'm going to add in some ideas tomorrow/Monday too to see how I can achieve this and keep my mobile toolbar.

**Comment by dev-palwar** [comment]
We just need to add a button that scrolls the page back to top its not that deep 😅

**Comment by NiallJoeMaher** [comment]
@dev-palwar, you might not think it! But as a solo maintainer, I have to consider these things for UX/accessibility and preferences. 

We have a mobile navbar already, so adding another floating button might be weird. 

![Screenshot 2023-10-22 at 08 43 57](https://github.com/codu-code/codu/assets/12615742/839685bb-9f9f-4b60-81f8-ce14cebc3a3a)

But maybe there is an option to incorporate it into the existing menu. 🤔

**Comment by Bitnagar** [comment]
@NiallJoeMaher CSS Tricks do it this way. Just a small size button.

![image](https://github.com/codu-code/codu/assets/88842947/51c57a37-8b41-449a-bbb1-915331cd57ad)

**Comment by garyb1** [comment]
Certainly, several factors need to be taken into account. @NiallJoeMaher is right to take time to make sure it is right. 

Nielson/Norman Group have some good advice on [back to top](https://www.nngroup.com/articles/back-to-top/). Some notes include deciding if a back to top is needed. The [Scottish Government DS team](https://designsystem.gov.scot/components/back-to-top) referred to that in their back to top implementation.

Back to Top is more of a convenience for mouse or touch users. For  users relying on keyboard or assistive tech, there are already existing shortcuts and key commands to achieve the same effect, making it less of a concern. For example, keyboard shortcuts like `ctrl + home` or just `home` (`fn + left/right` on Mac) will work. 

In the DOM it might live above or below the footer info. Most examples i've seen live above footer.

Deciding where to send focus will be important here. An anchor element would be perfect. But should it navigate to the top (or to first focusable element (skip link if exists)), or top of the main content area. I think top of the page (or skip link) would be fine to be honest. 

A visual label with "Back to Top" would be much better for many users. An icon on it's own might be ambiguous.

### Accommodating smooth scrolling
Smooth scrolling should only be employed when the user has no reduced motion preference enabled.

```
@media (prefers-reduced-motion: no-preference) {
    scroll-behavior: smooth;
 } 
```
